File: 2800-minimum-string-length-after-removing-substrings/minimum-string-length-after-removing-substrings.java
class Solution {
    public int minLength(String s) {
        Stack<Character> stack = new Stack<>();
        for (int i = 0; i < s.length(); i++){
            char c = s.charAt(i);
            if (stack.isEmpty()){
                stack.push(c);
                continue;
            }
            if (c == 'B' && stack.peek() == 'A'){
                stack.pop();
            }
            else if (c == 'D' && stack.peek() == 'C'){
                stack.pop();
            }
            else{
                stack.add(c);
            }
        }
        return stack.size();
    }
}

File: 2800-minimum-string-length-after-removing-substrings/minimum-string-length-after-removing-substrings.py
class Solution:
    def minLength(self, s: str) -> int:
        stack = []
        for c in s:
            if not stack:
                stack.append(c)
                continue
            if stack[-1] == 'A' and c == 'B':
                stack.pop()
            elif stack[-1] == 'C' and c == 'D':
                stack.pop()
            else:
                stack.append(c)
        return len(stack)